### PR TITLE
Add customizable Custom Games

### PR DESCRIPTION
**Pull request by TheImmortal**

### CODE REVIEW

**Comment by Zarel** [comment]
I think those build errors have to do with the way you're changing the formats. You probably need to clone/deep-clone some objects you're currently taking references from.

**Comment by Zarel** [comment]
Wait, you already do. Where is this build failure coming from?

**Comment by Zarel** [comment]
Maybe you need to explicitly interpret `customgame` as `customgame0`?

**Comment by TheImmortal** [comment]
I figured it out. It was the additional ``battle.commitDecisions();`` in tests that took Team Preview into account.

**Comment by Zarel** [comment]
It should probably be one commit; I think commits should be atomic: unit tests should not fail between commits. My approach has always been to change APIs and how the API is used in the same commit.

**Comment by Zarel** [comment]
Also, I wonder if maybe Team Preview should be on by default...

**Comment by TheImmortal** [comment]
Pretty sure ascriptmaster complained about Team Preview in tests a while back. They shouldn't be on by default imo.

**Comment by Zarel** [comment]
It makes sense that following game mechanics should be on by default, or at least all follow a single flag `customgame1`.

**Comment by TheImmortal** [comment]
You can have it checked by default on the client display, nothing here needs to be on.

**Comment by TheImmortal** [comment]
Or were you referring to tests following game mechanics?

**Comment by Zarel** [comment]
It's really weird for the default custom game setting to be `customgame5`. I think it should either be `0` or `1`...

File: tools.js [hunk 1]
```diff
@@ -422,6 +429,38 @@ module.exports = (() => {
...
+					// Team Preview
+					{option: 'Team Preview', type: 'ruleset'},
+					// Sleep Clause Mod
+					{option: 'Sleep Clause Mod', type: 'ruleset'},
+				];
+
+				customGame = parseInt(customGame[2]);
+				for (let i = 0; i < options.length; i++) {
+					if (!(customGame & (1 << i))) continue;
+					effect[options[i].type] = effect[options[i].type].concat(options[i].option);
```
Comment (kotarou3) [inline tools.js]: As opposed to `#push()`? Might need to add a small comment explaining why you did it this way

File: tools.js [hunk 2]
```diff
@@ -422,6 +429,38 @@ module.exports = (() => {
...
+					{option: ['Illegal', 'Unreleased'], type: 'banlist'},
+					// Team Preview
+					{option: 'Team Preview', type: 'ruleset'},
+					// Sleep Clause Mod
+					{option: 'Sleep Clause Mod', type: 'ruleset'},
+				];
+
+				customGame = parseInt(customGame[2]);
+				for (let i = 0; i < options.length; i++) {
+					if (!(customGame & (1 << i))) continue;
```
Comment (kotarou3) [inline tools.js]: Feels strange seeing bit level masking for flags when we've always done it with explicit flag names

**Comment by ascriptmaster** [comment]
I don't care either way when it comes to team preview on as default or not. The testing issue wasn't really too big of a deal anyways, it was just a minor gripe

**Comment by ascriptmaster** [comment]
upd8 wen

**Comment by Zarel** [comment]
All in due time...

**Comment by Zarel** [comment]
So now we have a `customRules` system which supersedes this... sorry for delaying this so long :<

**Comment by TheImmortal** [comment]
So when do we expect the client side update?

**Comment by Zarel** [comment]
Blaaaah my to-do list is _huge_ though... Feel free to add an issue for it somewhere.

**Pull request closed without merging**